test/test_genome.py

```diff
@@ -843,9 +843,25 @@ def test_set_expression_for_output(genome_params, rng):
     assert CartesianGraph(genome).to_sympy() == x_0 + x_1
 
     new_dna = [1, 0, 1]
-    genome.set_expression_for_output(new_dna)
+    genome.set_expression_for_output(dna_insert=new_dna, target_expression=" x_0 - x_1")
     assert CartesianGraph(genome).to_sympy() == x_0 - x_1
 
     new_dna = [0, 0, 1, 2, 0, 0, 1, 0, 0, 0, 2, 3]  # x_0+x_1; 1.0; 0; x_0+x_1 + 1.0
-    genome.set_expression_for_output(new_dna)
+    genome.set_expression_for_output(dna_insert=new_dna, target_expression=" x_0 + x_1 + 1.0")
     assert CartesianGraph(genome).to_sympy() == x_0 + x_1 + 1.0
+
+    with pytest.raises(ValueError):
+        genome.set_expression_for_output(dna_insert=new_dna, target_expression=" x_0 + x_1 + 1")
```

Comment:
hmm this test needs a comment, otherwise it's pretty surprising that it's expected to fail.

in addition i would suggest to add an expected-failure test where the expression are indeed different, for example by replacing `+` with `8`.